dbus/xdg.c

```diff
@@ -109,6 +111,7 @@ static int handle_notify(sd_bus_message *msg, void *data,
 	notif->app_icon = strdup(app_icon);
 	notif->summary = strdup(summary);
 	notif->body = strdup(body);
+	notif->icon = create_icon(notif->app_icon, state->config.superstyle.max_icon_size);
```

Comment:
Is it possible to check for `style->icons` here instead of in the rendering functions? This would allow to reduce memory usage in case icons are disabled. The downside is that dynamically changing the configuration to enable icons won't draw icons for existing notifications, but I think that's fine.

Reply:
Yes, that should work. It makes the checks in `render.c` nicer too. Good catch.